gravitee-apim-console-webui/src/management/settings/custom-user-fields/migrated/custom-user-fields-migrated.component.ts

```diff
@@ -0,0 +1,215 @@
+    });
+  }
+
+  private sortFields({ active, direction }: Sort): CustomUserField[] {
+    if (direction === 'asc' || direction === 'desc') {
+      return orderBy(this.customUserFieldsFiltered, [(item) => get(item, active).toLowerCase()], direction);
+    }
+
+    return this.customUserFieldsFiltered;
+  }
+
+  private filterFields(searchTerm: string): CustomUserField[] {
+    return this.customUserFields.filter((customUserField: CustomUserField) => {
+      return (
+        customUserField.key.toLowerCase().includes(searchTerm) ||
+        customUserField.label.toLowerCase().includes(searchTerm) ||
+        this.searchValues(customUserField.values, searchTerm) ||
+        this.searchRequired(customUserField.required, searchTerm)
+      );
+    });
+  }
+
+  private paginate({ index, size }) {
+   return this.customUserFieldsFiltered.slice((index - 1) * size, index * size);
+  }
+
+  private searchValues(values: string[], searchTerm: string): boolean {
+    if (!values || values.length < 1) {
+      return false;
+    }
+    return values.join(' ').toLowerCase().includes(searchTerm);
+  }
+
+  private searchRequired(isRequired: boolean, searchTerm: string): boolean {
+    if (!isRequired) {
+      return false;
+    }
+    return 'required'.includes(searchTerm);
+  }
+
+  public create() {
+    this.matDialog
+      .open<CustomUserFieldsDialogComponent, CustomUserFieldsDialogData, CustomUserFieldsDialogResult>(CustomUserFieldsDialogComponent, {
+        data: {
+          action: 'Create',
+        },
+        width: GIO_DIALOG_WIDTH.SMALL,
+      })
+      .afterClosed()
+      .pipe(
+        filter((dialogResult) => !!dialogResult),
+        switchMap((dialogResult) => {
+          this.isLoading = true;
+          return this.customUserFieldsService.create(dialogResult);
+        }),
+      )
+      .subscribe({
+        next: (customUserField) => this.initList(`Field ${customUserField.key} created successfully`),
+        error: ({ error }) => this.snackBarService.error(error?.message ? error.message : 'Error during field creation!'),
+      })
+      .add(() => (this.isLoading = false));
+  }
+
+  public update(customUserField: CustomUserField) {
+    this.matDialog
+      .open<CustomUserFieldsDialogComponent, CustomUserFieldsDialogData, CustomUserFieldsDialogResult>(CustomUserFieldsDialogComponent, {
+        data: {
+          action: 'Update',
+          ...customUserField,
+        },
+        width: GIO_DIALOG_WIDTH.SMALL,
+      })
+      .afterClosed()
+      .pipe(
+        filter((data) => !!data),
+        switchMap((data) => {
+          this.isLoading = true;
+          return this.customUserFieldsService.update(data);
+        }),
+      )
+      .subscribe({
+        next: () => this.initList(`Field ${customUserField.key} updated successfully`),
+        error: ({ error }) => this.snackBarService.error(error?.message ? error.message : 'Error during field update!'),
+      })
+      .add(() => (this.isLoading = false));
```

Comment:
IMHO. with my suggesting approach you need only one boolean for loading state. 

at the start of a refresh event 
- reset the loading state to true  + empty the list from the customUserFieldsFiltered
- make a new call to the list 
- then change the loading to false